src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -1002,4 +1291,14 @@ public SearchResultPageDTO<Exam> getAllOnPageWithSize(final PageableSearchDTO<St
         }
         return new SearchResultPageDTO<>(examPage.getContent(), examPage.getTotalPages());
     }
+
+    /**
+     * A specialized BiFunction<Long, Double, BonusResultDTO> functional interface to provide a simple interface
+     * for passing the data dependencies needed for a Bonus calculation (like source course/exam results).
+     */
+    @FunctionalInterface
+    private interface ExamBonusCalculator {
+
+        BonusResultDTO calculateStudentGradesWithBonus(Long studentId, Double achievedPointsOfBonusTo);
```

Comment:
😍 Nice use of functional programming.

Reply:
Thanks 😄 I also think this version looks much more elegant than my first attempt where I tried passing a complex `Map` with all needed data.